dns.c

```diff
@@ -25,8 +25,8 @@ extern int res_search();
 static unsigned short getshort(c) unsigned char *c;
 { unsigned short u; u = c[0]; return (u << 8) + c[1]; }
 
-static struct { unsigned char *buf; } response;
-static int responsebuflen = 0;
+static char *responsebuf;
```

Comment:
You change this from ```unsigned char *``` to ```char *```. And at several places below this is assigned from an ```unsigned char *``` variable (when being allocated) or passed as parameter to ```res_query``` or ```res_search```, which expect an ```unsigned char *```. The latter is hidden because the ```lookup``` function pointer does not declare parameter types…

Reply:
Back to unsigned then. One example showing that expanding prototypes helps with collaboration.